crunchers/cruncher_top_cards.cc

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************************
+ The MIT License (MIT)
+
+ Copyright (c) 2016 Grigory Nikolaenko <nikolaenko.grigory@gmail.com>
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ *******************************************************************************/
+
+#include "cruncher_top_cards.h"
+#include "schema.h"
+
+#include "../../Current/Bricks/dflags/dflags.h"
+#include "../../Current/Sherlock/replicator.h"
+
+namespace CTFO {
+
+CURRENT_STRUCT(TopCardsSubCruncherParams) {
+  CURRENT_FIELD(interval, std::chrono::microseconds);
+  CURRENT_FIELD_DESCRIPTION(interval, "Size of the cruncher sliding window.");
+  CURRENT_FIELD(top_size, uint64_t);
+  CURRENT_FIELD_DESCRIPTION(top_size, "Maximum number of items in the top cards list.");
+  CURRENT_DEFAULT_CONSTRUCTOR(TopCardsSubCruncherParams) {}
+  CURRENT_CONSTRUCTOR(TopCardsSubCruncherParams)(std::chrono::microseconds interval, uint64_t top_size)
+      : interval(interval), top_size(top_size) {}
+};
+
+CURRENT_STRUCT(TopCardsCruncherParams) {
+  CURRENT_FIELD(remote_url, std::string, "http://localhost:8383/raw_log");
+  CURRENT_FIELD_DESCRIPTION(remote_url, "Remote CTFOServer URL to subscribe to.");
+  CURRENT_FIELD(route, std::string, "/top_cards");
+  CURRENT_FIELD_DESCRIPTION(route, "The HTTP url path to expose the cruncher in.");
+  CURRENT_FIELD(port, uint16_t, 8374);
+  CURRENT_FIELD_DESCRIPTION(port, "Port to spawn the cruncher on.");
+  CURRENT_FIELD(args, std::vector<TopCardsSubCruncherParams>);
+  CURRENT_FIELD_DESCRIPTION(args, "List of params for several individual cruncher instances.");
+  CURRENT_DEFAULT_CONSTRUCTOR(TopCardsCruncherParams)
+      : args{TopCardsSubCruncherParams(std::chrono::seconds(10), 3),
+             TopCardsSubCruncherParams(std::chrono::hours(1), 5),
+             TopCardsSubCruncherParams(std::chrono::hours(24), 10)} {}
+};
+
+}  // namespace CTFO
+
+using namespace CTFO;
```

Comment:
Given you do specify `CTFO::` directly, perhaps get rid of this top-level `using`?